### PR TITLE
BICAWS7-1465-Automatically-Format-ASN-Input-Field

### PR DESCRIPTION
**Pull request by Senny10**
As a user I would like to correct an ASN error using a input field that auto-formats my input into segments so I can easily see each section of the code (slashes / in the box separating sections of characters which automatically jumps you to the next section)
image

### CODE REVIEW

File: src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx [hunk 1]
```diff
@@ -146,24 +162,27 @@ export const DefendantDetails = ({ stopLeavingFn }: DefendantDetailsProps) => {
           >
             <Label>{"Enter the ASN"}</Label>
             <HintText>{"Long form ASN"}</HintText>
```
Comment (richrace) [inline src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx:164]: Can you enter long form ASN?

File: src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx [hunk 2]
```diff
@@ -95,10 +110,11 @@ export const DefendantDetails = ({ stopLeavingFn }: DefendantDetailsProps) => {
 
   const handleAsnChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
     const asn = e.target.value.toUpperCase()
-    setIsValidAsn(isAsnFormatValid(asn))
+    const sanitized = asn.replace(/[^\dA-Z]/gi, "").toUpperCase()
```
Comment (richrace) [inline src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx:113]: What does this do?

Suggestion: extract to a function with better description

File: src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx [hunk 3]
```diff
@@ -43,6 +43,21 @@ const useStyles = createUseStyles({
   }
 })
 
+const formatAsn = (input: string) => {
+  const sanitized = input.replace(/[^\dA-Z]/gi, "").toUpperCase()
```
Comment (richrace) [inline src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx:47]: Duplication

File: src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx (hunk 1 above)
Comment (Senny10) [reply src/features/CourtCaseDetails/Tabs/Panels/DefendantDetails.tsx:164]: Not yet, I have to work on that.

**Pull request closed without merging**